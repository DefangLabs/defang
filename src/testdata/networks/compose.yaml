services:
  service-invalid:
    image: example
    networks:
      invalid-network-name: {}
  service-public:
    image: example
    networks:
      public: {}
  service-public-list:
    image: example
    networks:
      - public
  service-private:
    image: example
    networks:
      - private
  service-internal:
    image: example
    networks:
      - internal
  service-default:
    image: example
    networks:
      - default
  service-multi:
    image: example
    networks:
      - default
      - internal
networks:
  public:
    external: true
  internal:
    internal: true
