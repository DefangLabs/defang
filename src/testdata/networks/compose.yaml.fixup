{
  "service-default": {
    "command": null,
    "entrypoint": null,
    "image": "example",
    "networks": {
      "default": null
    }
  },
  "service-internal": {
    "command": null,
    "entrypoint": null,
    "image": "example",
    "networks": {
      "internal": null
    }
  },
  "service-invalid": {
    "command": null,
    "entrypoint": null,
    "image": "example",
    "networks": {
      "invalid-network-name": {}
    }
  },
  "service-multi": {
    "command": null,
    "entrypoint": null,
    "image": "example",
    "networks": {
      "default": null,
      "internal": null
    }
  },
  "service-private": {
    "command": null,
    "entrypoint": null,
    "image": "example",
    "networks": {
      "private": null
    }
  },
  "service-public": {
    "command": null,
    "entrypoint": null,
    "image": "example",
    "networks": {
      "public": {}
    }
  },
  "service-public-list": {
    "command": null,
    "entrypoint": null,
    "image": "example",
    "networks": {
      "public": null
    }
  }
}