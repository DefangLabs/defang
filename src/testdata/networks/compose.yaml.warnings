 ! service "service1": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "service2": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "service3": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "service4": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors