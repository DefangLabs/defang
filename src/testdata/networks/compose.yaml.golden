name: networks
services:
  service-default:
    image: example
    networks:
      default: null
  service-internal:
    image: example
    networks:
      internal: null
  service-invalid:
    image: example
    networks:
      invalid-network-name: {}
  service-multi:
    image: example
    networks:
      default: null
      internal: null
  service-private:
    image: example
    networks:
      private: null
  service-public:
    image: example
    networks:
      public: {}
  service-public-list:
    image: example
    networks:
      public: null
networks:
  default:
    name: networks_default
  internal:
    name: networks_internal
    internal: true
  public:
    name: public
    external: true
